sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2729,6 +2749,18 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as an ArrowRecordBatch, and serve the ArrowRecordBatch to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payload = collectAsArrow()
+    val payloadBytes = ArrowConverters.payloadToByteArray(payload, this.schema)
+
+    withNewExecutionId {
+      PythonRDD.serveIterator(Iterator(payloadBytes), "serve-Arrow")
```

Comment:
Thanks @viirya for the review, I'm working on another update that will do conversion on the executors